{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "a3c5bfc4",
   "metadata": {},
   "outputs": [],
   "source": [
    "import duckdb\n",
    "from dotenv import load_dotenv\n",
    "import os\n",
    "import pandas as pd\n",
    "\n",
    "load_dotenv()  # loads variables from .env\n",
    "db_path = os.getenv(\"DB_PATH_sql\")\n",
    "mtrain = os.getenv(\"DATA_TRAIN_PATH\")\n",
    "\n",
    "pd.set_option('display.max_rows', None)\n",
    "pd.set_option('display.max_columns', None)\n",
    "\n",
    "# Optional: prevent truncation of long strings\n",
    "pd.set_option('display.max_colwidth', None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "48c5a6b0",
   "metadata": {},
   "outputs": [],
   "source": [
    "c1 = duckdb.connect(db_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "4249b42d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Table: allele\n",
      "  - AlleleID (BIGINT)\n",
      "  - HgvsNotation (VARCHAR)\n",
      "  - GeneID (BIGINT)\n",
      "  - HGNCID (VARCHAR)\n",
      "  - ClinicalSignificance (VARCHAR)\n",
      "  - LastEvaluated (VARCHAR)\n",
      "  - RS# (dbSNP) (BIGINT)\n",
      "  - RCVAccession (VARCHAR)\n",
      "  - PhenotypeIDS (VARCHAR)\n",
      "  - PhenotypeList (VARCHAR)\n",
      "  - Origin (VARCHAR)\n",
      "  - ChromosomeAccession (VARCHAR)\n",
      "  - Chromosome (VARCHAR)\n",
      "  - ReviewStatus (VARCHAR)\n",
      "  - NumberSubmitters (BIGINT)\n",
      "  - TestedInGTR (VARCHAR)\n",
      "  - OtherIDs (VARCHAR)\n",
      "  - VariationID (BIGINT)\n",
      "  - PositionVCF (BIGINT)\n",
      "  - ReferenceAlleleVCF (VARCHAR)\n",
      "  - AlternateAlleleVCF (VARCHAR)\n",
      "  - GenesPerAlleleID (DOUBLE)\n",
      "  - Category (VARCHAR)\n",
      "  - MC (VARCHAR)\n",
      "\n",
      "Table: cross_references\n",
      "  - AlleleID (BIGINT)\n",
      "  - Database (VARCHAR)\n",
      "  - ID (VARCHAR)\n",
      "  - LastUpdated (VARCHAR)\n",
      "\n",
      "Table: gene\n",
      "  - GeneID (BIGINT)\n",
      "  - GeneSymbol (VARCHAR)\n",
      "  - GeneName (VARCHAR)\n",
      "  - GeneLevelDisease (VARCHAR)\n",
      "  - TotalSubmissions (DOUBLE)\n",
      "  - TotalAlleles (DOUBLE)\n",
      "  - SubmissionsReportingThisGene (DOUBLE)\n",
      "  - AllelesReportedPathogenicLikelyPathogenic (DOUBLE)\n",
      "  - NumberUncertain (DOUBLE)\n",
      "  - NumberWithConflicts (DOUBLE)\n",
      "\n",
      "Table: hgvs4variation\n",
      "  - VariationID (BIGINT)\n",
      "  - AlleleID (BIGINT)\n",
      "  - NucleotideExpression (VARCHAR)\n",
      "  - NucleotideChange (VARCHAR)\n",
      "\n",
      "Table: organization_summary\n",
      "  - OrganizationName (VARCHAR)\n",
      "  - OrganizationID (BIGINT)\n",
      "  - InstitutionType (VARCHAR)\n",
      "  - NumberOfClinVarSubmissions (BIGINT)\n",
      "  - MaximumReviewStatus (VARCHAR)\n",
      "  - CollectionMethods (VARCHAR)\n",
      "  - ClinicalSignificanceCategoriesSubmitted (VARCHAR)\n",
      "  - NumberOfSubmissionsFromClinicalTesting (BIGINT)\n",
      "  - NumberOfSubmissionsFromResearch (BIGINT)\n",
      "  - NumberOfSubmissionsFromLiteratureOnly (BIGINT)\n",
      "  - NumberOfSubmissionsFromCuration (BIGINT)\n",
      "  - NumberOfSubmissionsFromPhenotyping (BIGINT)\n",
      "\n",
      "Table: submission_summary\n",
      "  - VariationID (BIGINT)\n",
      "  - ClinicalSignificance (VARCHAR)\n",
      "  - Description (VARCHAR)\n",
      "  - SubmittedPhenotypeInfo (VARCHAR)\n",
      "  - ReportedPhenotypeInfo (VARCHAR)\n",
      "  - ReviewStatus (VARCHAR)\n",
      "  - CollectionMethod (VARCHAR)\n",
      "  - OriginCounts (VARCHAR)\n",
      "  - Submitter (VARCHAR)\n",
      "  - SubmittedGeneSymbol (VARCHAR)\n",
      "  - ExplanationOfInterpretation (VARCHAR)\n",
      "  - ContributesToAggregateClassification (VARCHAR)\n",
      "\n",
      "Table: summary_of_conflicting_interpretations\n",
      "  - VariationID (BIGINT)\n",
      "  - ClinvarPreferred (VARCHAR)\n",
      "  - Submitter1 (VARCHAR)\n",
      "  - Submitter1Clinsig (VARCHAR)\n",
      "  - Submitter1Reviewstatus (VARCHAR)\n",
      "  - Submitter1Description (VARCHAR)\n",
      "  - Submitter2 (VARCHAR)\n",
      "  - Submitter2Clinsig (VARCHAR)\n",
      "  - Submitter2Reviewstatus (VARCHAR)\n",
      "  - Submitter2Description (VARCHAR)\n",
      "  - RankDiff (BIGINT)\n",
      "  - ConflictReported (VARCHAR)\n",
      "  - VariantType (VARCHAR)\n",
      "  - Submitter1Method (VARCHAR)\n",
      "  - Submitter2Method (VARCHAR)\n",
      "\n",
      "Table: var_citations\n",
      "  - AlleleID (BIGINT)\n",
      "  - VariationID (BIGINT)\n",
      "  - CitationSource (VARCHAR)\n",
      "  - CitationID (VARCHAR)\n",
      "  - OrganizationID (VARCHAR)\n"
     ]
    }
   ],
   "source": [
    "tables = c1.execute(\"SELECT name FROM sqlite_master WHERE type='table';\").fetchall()\n",
    "\n",
    "for table in tables:\n",
    "    table_name = table[0]\n",
    "    print(f\"\\nTable: {table_name}\")\n",
    "    # Get column info\n",
    "    columns = c1.execute(f\"PRAGMA table_info({table_name});\").fetchall()\n",
    "    for col in columns:\n",
    "        print(f\"  - {col[1]} ({col[2]})\")  # col[1] = column name, col[2] = type\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "7c660232",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AlleleID</th>\n",
       "      <th>HgvsNotation</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>HGNCID</th>\n",
       "      <th>ClinicalSignificance</th>\n",
       "      <th>LastEvaluated</th>\n",
       "      <th>RS# (dbSNP)</th>\n",
       "      <th>RCVAccession</th>\n",
       "      <th>PhenotypeIDS</th>\n",
       "      <th>PhenotypeList</th>\n",
       "      <th>Origin</th>\n",
       "      <th>ChromosomeAccession</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>ReviewStatus</th>\n",
       "      <th>NumberSubmitters</th>\n",
       "      <th>TestedInGTR</th>\n",
       "      <th>OtherIDs</th>\n",
       "      <th>VariationID</th>\n",
       "      <th>PositionVCF</th>\n",
       "      <th>ReferenceAlleleVCF</th>\n",
       "      <th>AlternateAlleleVCF</th>\n",
       "      <th>GenesPerAlleleID</th>\n",
       "      <th>Category</th>\n",
       "      <th>MC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>15043</td>\n",
       "      <td>NM_014630.3(ZNF592):c.3136G&gt;A (p.Gly1046Arg)</td>\n",
       "      <td>9640</td>\n",
       "      <td>HGNC:28986</td>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>Jun 29, 2015</td>\n",
       "      <td>150829393</td>\n",
       "      <td>RCV000000014</td>\n",
       "      <td>MONDO:MONDO:0033005,MedGen:C4551772,OMIM:251300,Orphanet:2065,Orphanet:83472</td>\n",
       "      <td>Galloway-Mowat syndrome 1</td>\n",
       "      <td>germline</td>\n",
       "      <td>NC_000015.10</td>\n",
       "      <td>15</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>1</td>\n",
       "      <td>N</td>\n",
       "      <td>ClinGen:CA210674,UniProtKB:Q92610#VAR_064583,OMIM:613624.0001</td>\n",
       "      <td>4</td>\n",
       "      <td>84799209</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>within single gene</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>15044</td>\n",
       "      <td>NM_017547.4(FOXRED1):c.694C&gt;T (p.Gln232Ter)</td>\n",
       "      <td>55572</td>\n",
       "      <td>HGNC:26927</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Mar 12, 2024</td>\n",
       "      <td>267606829</td>\n",
       "      <td>RCV000000015|RCV000578659|RCV001194045|RCV003390625</td>\n",
       "      <td>MONDO:MONDO:0032624,MedGen:C4748791,OMIM:618241|MedGen:C3661900|MONDO:MONDO:0009723,MedGen:C2931891,OMIM:256000,Orphanet:506|</td>\n",
       "      <td>Mitochondrial complex 1 deficiency, nuclear type 19|not provided|Leigh syndrome|FOXRED1-related disorder</td>\n",
       "      <td>germline</td>\n",
       "      <td>NC_000011.10</td>\n",
       "      <td>11</td>\n",
       "      <td>criteria provided, multiple submitters, no conflicts</td>\n",
       "      <td>6</td>\n",
       "      <td>N</td>\n",
       "      <td>ClinGen:CA113792,OMIM:613622.0001</td>\n",
       "      <td>5</td>\n",
       "      <td>126275389</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1.0</td>\n",
       "      <td>within single gene</td>\n",
       "      <td>SO:0001587|nonsense,SO:0001619|non-coding_transcript_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>15045</td>\n",
       "      <td>NM_017547.4(FOXRED1):c.1289A&gt;G (p.Asn430Ser)</td>\n",
       "      <td>55572</td>\n",
       "      <td>HGNC:26927</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>Jun 06, 2024</td>\n",
       "      <td>267606830</td>\n",
       "      <td>RCV000000016</td>\n",
       "      <td>MONDO:MONDO:0032624,MedGen:C4748791,OMIM:618241</td>\n",
       "      <td>Mitochondrial complex 1 deficiency, nuclear type 19</td>\n",
       "      <td>germline</td>\n",
       "      <td>NC_000011.10</td>\n",
       "      <td>11</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>2</td>\n",
       "      <td>N</td>\n",
       "      <td>ClinGen:CA113794,UniProtKB:Q96CU9#VAR_064571,OMIM:613622.0002</td>\n",
       "      <td>6</td>\n",
       "      <td>126277517</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>within single gene</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001619|non-coding_transcript_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>15046</td>\n",
       "      <td>NM_025152.3(NUBPL):c.166G&gt;A (p.Gly56Arg)</td>\n",
       "      <td>80224</td>\n",
       "      <td>HGNC:20278</td>\n",
       "      <td>Conflicting classifications of pathogenicity</td>\n",
       "      <td>Nov 12, 2024</td>\n",
       "      <td>200401432</td>\n",
       "      <td>RCV000196589|RCV000622708|RCV001526454|RCV005055710</td>\n",
       "      <td>MedGen:C3661900|MeSH:D030342,MedGen:C0950123|MONDO:MONDO:0032625,MedGen:C4748792,OMIM:618242|MedGen:CN169374</td>\n",
       "      <td>not provided|Inborn genetic diseases|Mitochondrial complex 1 deficiency, nuclear type 21|not specified</td>\n",
       "      <td>germline;paternal</td>\n",
       "      <td>NC_000014.9</td>\n",
       "      <td>14</td>\n",
       "      <td>criteria provided, conflicting classifications</td>\n",
       "      <td>5</td>\n",
       "      <td>N</td>\n",
       "      <td>OMIM:613621.0001,ClinGen:CA321015,UniProtKB:Q8TB37#VAR_064570,ClinVar:7</td>\n",
       "      <td>214885</td>\n",
       "      <td>31562125</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>within single gene</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001619|non-coding_transcript_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>15048</td>\n",
       "      <td>NM_000410.4(HFE):c.845G&gt;A (p.Cys282Tyr)</td>\n",
       "      <td>3077</td>\n",
       "      <td>HGNC:4886</td>\n",
       "      <td>Pathogenic/Pathogenic, low penetrance; other; risk factor</td>\n",
       "      <td>Mar 04, 2025</td>\n",
       "      <td>1800562</td>\n",
       "      <td>RCV000000019|RCV000210820|RCV000308358|RCV000414811|RCV001270034|RCV001731264|RCV000178096|RCV001248830|RCV002280089|RCV002512585|RCV003224084|RCV003493406</td>\n",
       "      <td>MONDO:MONDO:0021001,MedGen:C3469186,OMIM:235200,Orphanet:139498,Orphanet:465508|MONDO:MONDO:0015356,MeSH:D009386,MedGen:C0027672,Orphanet:140162|MONDO:MONDO:0006507,MedGen:C0392514,OMIM:PS235200|Human Phenotype Ontology:HP:0010473,MedGen:C0151861;Human Phenotype Ontology:HP:0000992,Human Phenotype Ontology:HP:0005594,Human Phenotype Ontology:HP:0006831,Human Phenotype Ontology:HP:0007538,MONDO:MONDO:0005434,MedGen:C0349506|7 conditions|Human Phenotype Ontology:HP:0001638,MONDO:MONDO:0004994,MedGen:C0878544,Orphanet:167848|MedGen:C3661900|MedGen:C0018995||MeSH:D030342,MedGen:C0950123|6 conditions|MONDO:MONDO:0019257,MedGen:C0268060,Orphanet:79230</td>\n",
       "      <td>Hemochromatosis type 1|Hereditary cancer-predisposing syndrome|Hereditary hemochromatosis|Porphyrinuria;Cutaneous photosensitivity|7 conditions|Cardiomyopathy|not provided|Bronze diabetes|HFE-related disorder|Inborn genetic diseases|6 conditions|Juvenile hemochromatosis</td>\n",
       "      <td>biparental;germline;unknown</td>\n",
       "      <td>NC_000006.12</td>\n",
       "      <td>6</td>\n",
       "      <td>criteria provided, multiple submitters, no conflicts</td>\n",
       "      <td>53</td>\n",
       "      <td>Y</td>\n",
       "      <td>UniProtKB:Q30201#VAR_004398,OMIM:613609.0001,ClinGen:CA113795</td>\n",
       "      <td>9</td>\n",
       "      <td>26092913</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>within single gene</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001627|intron_variant</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   AlleleID                                  HgvsNotation  GeneID      HGNCID  \\\n",
       "0     15043  NM_014630.3(ZNF592):c.3136G>A (p.Gly1046Arg)    9640  HGNC:28986   \n",
       "1     15044   NM_017547.4(FOXRED1):c.694C>T (p.Gln232Ter)   55572  HGNC:26927   \n",
       "2     15045  NM_017547.4(FOXRED1):c.1289A>G (p.Asn430Ser)   55572  HGNC:26927   \n",
       "3     15046      NM_025152.3(NUBPL):c.166G>A (p.Gly56Arg)   80224  HGNC:20278   \n",
       "4     15048       NM_000410.4(HFE):c.845G>A (p.Cys282Tyr)    3077   HGNC:4886   \n",
       "\n",
       "                                        ClinicalSignificance LastEvaluated  \\\n",
       "0                                     Uncertain significance  Jun 29, 2015   \n",
       "1                                                 Pathogenic  Mar 12, 2024   \n",
       "2                                          Likely pathogenic  Jun 06, 2024   \n",
       "3               Conflicting classifications of pathogenicity  Nov 12, 2024   \n",
       "4  Pathogenic/Pathogenic, low penetrance; other; risk factor  Mar 04, 2025   \n",
       "\n",
       "   RS# (dbSNP)  \\\n",
       "0    150829393   \n",
       "1    267606829   \n",
       "2    267606830   \n",
       "3    200401432   \n",
       "4      1800562   \n",
       "\n",
       "                                                                                                                                                  RCVAccession  \\\n",
       "0                                                                                                                                                 RCV000000014   \n",
       "1                                                                                                          RCV000000015|RCV000578659|RCV001194045|RCV003390625   \n",
       "2                                                                                                                                                 RCV000000016   \n",
       "3                                                                                                          RCV000196589|RCV000622708|RCV001526454|RCV005055710   \n",
       "4  RCV000000019|RCV000210820|RCV000308358|RCV000414811|RCV001270034|RCV001731264|RCV000178096|RCV001248830|RCV002280089|RCV002512585|RCV003224084|RCV003493406   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    PhenotypeIDS  \\\n",
       "0                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   MONDO:MONDO:0033005,MedGen:C4551772,OMIM:251300,Orphanet:2065,Orphanet:83472   \n",
       "1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  MONDO:MONDO:0032624,MedGen:C4748791,OMIM:618241|MedGen:C3661900|MONDO:MONDO:0009723,MedGen:C2931891,OMIM:256000,Orphanet:506|   \n",
       "2                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                MONDO:MONDO:0032624,MedGen:C4748791,OMIM:618241   \n",
       "3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   MedGen:C3661900|MeSH:D030342,MedGen:C0950123|MONDO:MONDO:0032625,MedGen:C4748792,OMIM:618242|MedGen:CN169374   \n",
       "4  MONDO:MONDO:0021001,MedGen:C3469186,OMIM:235200,Orphanet:139498,Orphanet:465508|MONDO:MONDO:0015356,MeSH:D009386,MedGen:C0027672,Orphanet:140162|MONDO:MONDO:0006507,MedGen:C0392514,OMIM:PS235200|Human Phenotype Ontology:HP:0010473,MedGen:C0151861;Human Phenotype Ontology:HP:0000992,Human Phenotype Ontology:HP:0005594,Human Phenotype Ontology:HP:0006831,Human Phenotype Ontology:HP:0007538,MONDO:MONDO:0005434,MedGen:C0349506|7 conditions|Human Phenotype Ontology:HP:0001638,MONDO:MONDO:0004994,MedGen:C0878544,Orphanet:167848|MedGen:C3661900|MedGen:C0018995||MeSH:D030342,MedGen:C0950123|6 conditions|MONDO:MONDO:0019257,MedGen:C0268060,Orphanet:79230   \n",
       "\n",
       "                                                                                                                                                                                                                                                                    PhenotypeList  \\\n",
       "0                                                                                                                                                                                                                                                       Galloway-Mowat syndrome 1   \n",
       "1                                                                                                                                                                        Mitochondrial complex 1 deficiency, nuclear type 19|not provided|Leigh syndrome|FOXRED1-related disorder   \n",
       "2                                                                                                                                                                                                                             Mitochondrial complex 1 deficiency, nuclear type 19   \n",
       "3                                                                                                                                                                          not provided|Inborn genetic diseases|Mitochondrial complex 1 deficiency, nuclear type 21|not specified   \n",
       "4  Hemochromatosis type 1|Hereditary cancer-predisposing syndrome|Hereditary hemochromatosis|Porphyrinuria;Cutaneous photosensitivity|7 conditions|Cardiomyopathy|not provided|Bronze diabetes|HFE-related disorder|Inborn genetic diseases|6 conditions|Juvenile hemochromatosis   \n",
       "\n",
       "                        Origin ChromosomeAccession Chromosome  \\\n",
       "0                     germline        NC_000015.10         15   \n",
       "1                     germline        NC_000011.10         11   \n",
       "2                     germline        NC_000011.10         11   \n",
       "3            germline;paternal         NC_000014.9         14   \n",
       "4  biparental;germline;unknown        NC_000006.12          6   \n",
       "\n",
       "                                           ReviewStatus  NumberSubmitters  \\\n",
       "0                        no assertion criteria provided                 1   \n",
       "1  criteria provided, multiple submitters, no conflicts                 6   \n",
       "2                   criteria provided, single submitter                 2   \n",
       "3        criteria provided, conflicting classifications                 5   \n",
       "4  criteria provided, multiple submitters, no conflicts                53   \n",
       "\n",
       "  TestedInGTR  \\\n",
       "0           N   \n",
       "1           N   \n",
       "2           N   \n",
       "3           N   \n",
       "4           Y   \n",
       "\n",
       "                                                                  OtherIDs  \\\n",
       "0            ClinGen:CA210674,UniProtKB:Q92610#VAR_064583,OMIM:613624.0001   \n",
       "1                                        ClinGen:CA113792,OMIM:613622.0001   \n",
       "2            ClinGen:CA113794,UniProtKB:Q96CU9#VAR_064571,OMIM:613622.0002   \n",
       "3  OMIM:613621.0001,ClinGen:CA321015,UniProtKB:Q8TB37#VAR_064570,ClinVar:7   \n",
       "4            UniProtKB:Q30201#VAR_004398,OMIM:613609.0001,ClinGen:CA113795   \n",
       "\n",
       "   VariationID  PositionVCF ReferenceAlleleVCF AlternateAlleleVCF  \\\n",
       "0            4     84799209                  G                  A   \n",
       "1            5    126275389                  C                  T   \n",
       "2            6    126277517                  A                  G   \n",
       "3       214885     31562125                  G                  A   \n",
       "4            9     26092913                  G                  A   \n",
       "\n",
       "   GenesPerAlleleID            Category  \\\n",
       "0               1.0  within single gene   \n",
       "1               1.0  within single gene   \n",
       "2               1.0  within single gene   \n",
       "3               1.0  within single gene   \n",
       "4               1.0  within single gene   \n",
       "\n",
       "                                                                     MC  \n",
       "0                                           SO:0001583|missense_variant  \n",
       "1          SO:0001587|nonsense,SO:0001619|non-coding_transcript_variant  \n",
       "2  SO:0001583|missense_variant,SO:0001619|non-coding_transcript_variant  \n",
       "3  SO:0001583|missense_variant,SO:0001619|non-coding_transcript_variant  \n",
       "4                 SO:0001583|missense_variant,SO:0001627|intron_variant  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "c1.execute(\"SELECT * from allele limit 5\").fetchdf()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "17e6b2a8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AlleleID</th>\n",
       "      <th>HgvsNotation</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>HGNCID</th>\n",
       "      <th>ClinicalSignificance</th>\n",
       "      <th>LastEvaluated</th>\n",
       "      <th>RS# (dbSNP)</th>\n",
       "      <th>RCVAccession</th>\n",
       "      <th>PhenotypeIDS</th>\n",
       "      <th>PhenotypeList</th>\n",
       "      <th>Origin</th>\n",
       "      <th>ChromosomeAccession</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>ReviewStatus</th>\n",
       "      <th>NumberSubmitters</th>\n",
       "      <th>TestedInGTR</th>\n",
       "      <th>OtherIDs</th>\n",
       "      <th>VariationID</th>\n",
       "      <th>PositionVCF</th>\n",
       "      <th>ReferenceAlleleVCF</th>\n",
       "      <th>AlternateAlleleVCF</th>\n",
       "      <th>GenesPerAlleleID</th>\n",
       "      <th>Category</th>\n",
       "      <th>MC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>15045</td>\n",
       "      <td>NM_017547.4(FOXRED1):c.1289A&gt;G (p.Asn430Ser)</td>\n",
       "      <td>55572</td>\n",
       "      <td>HGNC:26927</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>Jun 06, 2024</td>\n",
       "      <td>267606830</td>\n",
       "      <td>RCV000000016</td>\n",
       "      <td>MONDO:MONDO:0032624,MedGen:C4748791,OMIM:618241</td>\n",
       "      <td>Mitochondrial complex 1 deficiency, nuclear type 19</td>\n",
       "      <td>germline</td>\n",
       "      <td>NC_000011.10</td>\n",
       "      <td>11</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>2</td>\n",
       "      <td>N</td>\n",
       "      <td>ClinGen:CA113794,UniProtKB:Q96CU9#VAR_064571,OMIM:613622.0002</td>\n",
       "      <td>6</td>\n",
       "      <td>126277517</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.0</td>\n",
       "      <td>within single gene</td>\n",
       "      <td>SO:0001583|missense_variant,SO:0001619|non-coding_transcript_variant</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   AlleleID                                  HgvsNotation  GeneID      HGNCID  \\\n",
       "0     15045  NM_017547.4(FOXRED1):c.1289A>G (p.Asn430Ser)   55572  HGNC:26927   \n",
       "\n",
       "  ClinicalSignificance LastEvaluated  RS# (dbSNP)  RCVAccession  \\\n",
       "0    Likely pathogenic  Jun 06, 2024    267606830  RCV000000016   \n",
       "\n",
       "                                      PhenotypeIDS  \\\n",
       "0  MONDO:MONDO:0032624,MedGen:C4748791,OMIM:618241   \n",
       "\n",
       "                                         PhenotypeList    Origin  \\\n",
       "0  Mitochondrial complex 1 deficiency, nuclear type 19  germline   \n",
       "\n",
       "  ChromosomeAccession Chromosome                         ReviewStatus  \\\n",
       "0        NC_000011.10         11  criteria provided, single submitter   \n",
       "\n",
       "   NumberSubmitters TestedInGTR  \\\n",
       "0                 2           N   \n",
       "\n",
       "                                                        OtherIDs  VariationID  \\\n",
       "0  ClinGen:CA113794,UniProtKB:Q96CU9#VAR_064571,OMIM:613622.0002            6   \n",
       "\n",
       "   PositionVCF ReferenceAlleleVCF AlternateAlleleVCF  GenesPerAlleleID  \\\n",
       "0    126277517                  A                  G               1.0   \n",
       "\n",
       "             Category  \\\n",
       "0  within single gene   \n",
       "\n",
       "                                                                     MC  \n",
       "0  SO:0001583|missense_variant,SO:0001619|non-coding_transcript_variant  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "c1.execute(\"SELECT * from allele where alleleId = 15045 limit 1\").fetchdf()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "7a992710",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>VariationID</th>\n",
       "      <th>ClinvarPreferred</th>\n",
       "      <th>Submitter1</th>\n",
       "      <th>Submitter1Clinsig</th>\n",
       "      <th>Submitter1Reviewstatus</th>\n",
       "      <th>Submitter1Description</th>\n",
       "      <th>Submitter2</th>\n",
       "      <th>Submitter2Clinsig</th>\n",
       "      <th>Submitter2Reviewstatus</th>\n",
       "      <th>Submitter2Description</th>\n",
       "      <th>RankDiff</th>\n",
       "      <th>ConflictReported</th>\n",
       "      <th>VariantType</th>\n",
       "      <th>Submitter1Method</th>\n",
       "      <th>Submitter2Method</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>734130</td>\n",
       "      <td>NC_000007.14:g.156791884T&gt;C</td>\n",
       "      <td>Labcorp Genetics (formerly Invitae), Labcorp</td>\n",
       "      <td>Benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>Not provided</td>\n",
       "      <td>PreventionGenetics, part of Exact Sciences</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>This variant is classified as likely benign based on ACMG/AMP sequence variant interpretation guidelines (Richards et al. 2015 PMID: 25741868, with internal and published modifications).</td>\n",
       "      <td>1</td>\n",
       "      <td>no</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>clinical testing</td>\n",
       "      <td>clinical testing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1344497</td>\n",
       "      <td>NC_012920.1(MT-CYB):m.235A&gt;G</td>\n",
       "      <td>Mendelics</td>\n",
       "      <td>Benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>This variant is considered likely benign or benign based on one or more of the following: it is predicted to be benign by multiple in silico algorithms, and/or has population frequency not consistent with disease, and/or has normal protein function, and/or has lack of segregation with disease, and/or has been detected in co-occurrence with known pathogenic variant, and/or has lack of disease association in case-control studies, and/or is located in a region inconsistent with a known cause of pathogenicity. GnomAD 4.1.0 frequency 0.04900 homoplsmic/9 heteroplasmic</td>\n",
       "      <td>Department of Pediatrics, Division of Medical Genetics, Faculty of Medicine Ramathibodi Hospital, Mahidol University</td>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>Not provided</td>\n",
       "      <td>2</td>\n",
       "      <td>yes</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>clinical testing</td>\n",
       "      <td>research</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>12321</td>\n",
       "      <td>NC_000008.11:g.11573132C&gt;T</td>\n",
       "      <td>OMIM</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>Not provided</td>\n",
       "      <td>H3Africa Consortium</td>\n",
       "      <td>Benign</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>While the frequency of the alternate allele in gnoMAD v2.0.2 is 0.146, its frequency in African populations is &gt;5%. This suggests that previous classifications of this variant as pathogenic are in error.</td>\n",
       "      <td>4</td>\n",
       "      <td>yes</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>literature only</td>\n",
       "      <td>research</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>12321</td>\n",
       "      <td>NC_000008.11:g.11573132C&gt;T</td>\n",
       "      <td>OMIM</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>Not provided</td>\n",
       "      <td>PreventionGenetics, part of Exact Sciences</td>\n",
       "      <td>Benign</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>This variant is classified as benign based on ACMG/AMP sequence variant interpretation guidelines (Richards et al. 2015 PMID: 25741868, with internal and published modifications).</td>\n",
       "      <td>4</td>\n",
       "      <td>yes</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>literature only</td>\n",
       "      <td>clinical testing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>12321</td>\n",
       "      <td>NC_000008.11:g.11573132C&gt;T</td>\n",
       "      <td>OMIM</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>Not provided</td>\n",
       "      <td>Reproductive Health Research and Development, BGI Genomics</td>\n",
       "      <td>Benign</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>NC_000008.11:g.11573132C&gt;T has an allele frequency of 0.163 in African subpopulation in the gnomAD database, including 128 homozygous occurrences. This variant was annotated at the position 11,468,050 on chromsome 8. In the functional study by MIN6 beta-cells, this variant decreased luciferase expression as compared with control constructs without any insert (PMID: 19667185). This evidence suggests the variant to be classified as benign. ACMG/AMP criteria applied: BA1, BS2, PS3.</td>\n",
       "      <td>4</td>\n",
       "      <td>yes</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>literature only</td>\n",
       "      <td>curation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>14669</td>\n",
       "      <td>NC_000010.11:g.6072697C&gt;A</td>\n",
       "      <td>OMIM</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>Not provided</td>\n",
       "      <td>PreventionGenetics, part of Exact Sciences</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>This variant is classified as likely benign based on ACMG/AMP sequence variant interpretation guidelines (Richards et al. 2015 PMID: 25741868, with internal and published modifications).</td>\n",
       "      <td>3</td>\n",
       "      <td>yes</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>literature only</td>\n",
       "      <td>clinical testing</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   VariationID              ClinvarPreferred  \\\n",
       "0       734130   NC_000007.14:g.156791884T>C   \n",
       "1      1344497  NC_012920.1(MT-CYB):m.235A>G   \n",
       "2        12321    NC_000008.11:g.11573132C>T   \n",
       "3        12321    NC_000008.11:g.11573132C>T   \n",
       "4        12321    NC_000008.11:g.11573132C>T   \n",
       "5        14669     NC_000010.11:g.6072697C>A   \n",
       "\n",
       "                                     Submitter1 Submitter1Clinsig  \\\n",
       "0  Labcorp Genetics (formerly Invitae), Labcorp            Benign   \n",
       "1                                     Mendelics            Benign   \n",
       "2                                          OMIM        Pathogenic   \n",
       "3                                          OMIM        Pathogenic   \n",
       "4                                          OMIM        Pathogenic   \n",
       "5                                          OMIM        Pathogenic   \n",
       "\n",
       "                Submitter1Reviewstatus  \\\n",
       "0  criteria provided, single submitter   \n",
       "1  criteria provided, single submitter   \n",
       "2       no assertion criteria provided   \n",
       "3       no assertion criteria provided   \n",
       "4       no assertion criteria provided   \n",
       "5       no assertion criteria provided   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       Submitter1Description  \\\n",
       "0                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               Not provided   \n",
       "1  This variant is considered likely benign or benign based on one or more of the following: it is predicted to be benign by multiple in silico algorithms, and/or has population frequency not consistent with disease, and/or has normal protein function, and/or has lack of segregation with disease, and/or has been detected in co-occurrence with known pathogenic variant, and/or has lack of disease association in case-control studies, and/or is located in a region inconsistent with a known cause of pathogenicity. GnomAD 4.1.0 frequency 0.04900 homoplsmic/9 heteroplasmic   \n",
       "2                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               Not provided   \n",
       "3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               Not provided   \n",
       "4                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               Not provided   \n",
       "5                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               Not provided   \n",
       "\n",
       "                                                                                                             Submitter2  \\\n",
       "0                                                                            PreventionGenetics, part of Exact Sciences   \n",
       "1  Department of Pediatrics, Division of Medical Genetics, Faculty of Medicine Ramathibodi Hospital, Mahidol University   \n",
       "2                                                                                                   H3Africa Consortium   \n",
       "3                                                                            PreventionGenetics, part of Exact Sciences   \n",
       "4                                                            Reproductive Health Research and Development, BGI Genomics   \n",
       "5                                                                            PreventionGenetics, part of Exact Sciences   \n",
       "\n",
       "        Submitter2Clinsig               Submitter2Reviewstatus  \\\n",
       "0           Likely benign       no assertion criteria provided   \n",
       "1  Uncertain significance       no assertion criteria provided   \n",
       "2                  Benign  criteria provided, single submitter   \n",
       "3                  Benign       no assertion criteria provided   \n",
       "4                  Benign       no assertion criteria provided   \n",
       "5           Likely benign       no assertion criteria provided   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 Submitter2Description  \\\n",
       "0                                                                                                                                                                                                                                                                                                           This variant is classified as likely benign based on ACMG/AMP sequence variant interpretation guidelines (Richards et al. 2015 PMID: 25741868, with internal and published modifications).   \n",
       "1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Not provided   \n",
       "2                                                                                                                                                                                                                                                                                          While the frequency of the alternate allele in gnoMAD v2.0.2 is 0.146, its frequency in African populations is >5%. This suggests that previous classifications of this variant as pathogenic are in error.   \n",
       "3                                                                                                                                                                                                                                                                                                                  This variant is classified as benign based on ACMG/AMP sequence variant interpretation guidelines (Richards et al. 2015 PMID: 25741868, with internal and published modifications).   \n",
       "4  NC_000008.11:g.11573132C>T has an allele frequency of 0.163 in African subpopulation in the gnomAD database, including 128 homozygous occurrences. This variant was annotated at the position 11,468,050 on chromsome 8. In the functional study by MIN6 beta-cells, this variant decreased luciferase expression as compared with control constructs without any insert (PMID: 19667185). This evidence suggests the variant to be classified as benign. ACMG/AMP criteria applied: BA1, BS2, PS3.   \n",
       "5                                                                                                                                                                                                                                                                                                           This variant is classified as likely benign based on ACMG/AMP sequence variant interpretation guidelines (Richards et al. 2015 PMID: 25741868, with internal and published modifications).   \n",
       "\n",
       "   RankDiff ConflictReported                VariantType  Submitter1Method  \\\n",
       "0         1               no  single nucleotide variant  clinical testing   \n",
       "1         2              yes  single nucleotide variant  clinical testing   \n",
       "2         4              yes  single nucleotide variant   literature only   \n",
       "3         4              yes  single nucleotide variant   literature only   \n",
       "4         4              yes  single nucleotide variant   literature only   \n",
       "5         3              yes  single nucleotide variant   literature only   \n",
       "\n",
       "   Submitter2Method  \n",
       "0  clinical testing  \n",
       "1          research  \n",
       "2          research  \n",
       "3  clinical testing  \n",
       "4          curation  \n",
       "5  clinical testing  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "c1.execute(\"SELECT * from summary_of_conflicting_interpretations limit 6\").fetchdf()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1350f356",
   "metadata": {},
   "source": [
    "# conflict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "ad3b42aa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AlleleID</th>\n",
       "      <th>HgvsNotation</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>HGNCID</th>\n",
       "      <th>ClinicalSignificance</th>\n",
       "      <th>LastEvaluated</th>\n",
       "      <th>RS# (dbSNP)</th>\n",
       "      <th>RCVAccession</th>\n",
       "      <th>PhenotypeIDS</th>\n",
       "      <th>PhenotypeList</th>\n",
       "      <th>Origin</th>\n",
       "      <th>ChromosomeAccession</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>ReviewStatus</th>\n",
       "      <th>NumberSubmitters</th>\n",
       "      <th>TestedInGTR</th>\n",
       "      <th>OtherIDs</th>\n",
       "      <th>VariationID</th>\n",
       "      <th>PositionVCF</th>\n",
       "      <th>ReferenceAlleleVCF</th>\n",
       "      <th>AlternateAlleleVCF</th>\n",
       "      <th>GenesPerAlleleID</th>\n",
       "      <th>Category</th>\n",
       "      <th>MC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>27360</td>\n",
       "      <td>NC_000008.11:g.11573132C&gt;T</td>\n",
       "      <td>640</td>\n",
       "      <td>HGNC:1057</td>\n",
       "      <td>Benign</td>\n",
       "      <td>Oct 28, 2020</td>\n",
       "      <td>61199332</td>\n",
       "      <td>RCV000013114|RCV001777135|RCV003914831</td>\n",
       "      <td>MONDO:MONDO:0013242,MedGen:C3150618,OMIM:613375,Orphanet:552|MedGen:CN169374|</td>\n",
       "      <td>Maturity-onset diabetes of the young type 11|not specified|BLK-related disorder</td>\n",
       "      <td>germline</td>\n",
       "      <td>NC_000008.11</td>\n",
       "      <td>8</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>4</td>\n",
       "      <td>N</td>\n",
       "      <td>ClinGen:CA10602353,OMIM:191305.0003</td>\n",
       "      <td>12321</td>\n",
       "      <td>11573132</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1.0</td>\n",
       "      <td>asserted, but not computed</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   AlleleID                HgvsNotation  GeneID     HGNCID  \\\n",
       "0     27360  NC_000008.11:g.11573132C>T     640  HGNC:1057   \n",
       "\n",
       "  ClinicalSignificance LastEvaluated  RS# (dbSNP)  \\\n",
       "0               Benign  Oct 28, 2020     61199332   \n",
       "\n",
       "                             RCVAccession  \\\n",
       "0  RCV000013114|RCV001777135|RCV003914831   \n",
       "\n",
       "                                                                    PhenotypeIDS  \\\n",
       "0  MONDO:MONDO:0013242,MedGen:C3150618,OMIM:613375,Orphanet:552|MedGen:CN169374|   \n",
       "\n",
       "                                                                     PhenotypeList  \\\n",
       "0  Maturity-onset diabetes of the young type 11|not specified|BLK-related disorder   \n",
       "\n",
       "     Origin ChromosomeAccession Chromosome  \\\n",
       "0  germline        NC_000008.11          8   \n",
       "\n",
       "                          ReviewStatus  NumberSubmitters TestedInGTR  \\\n",
       "0  criteria provided, single submitter                 4           N   \n",
       "\n",
       "                              OtherIDs  VariationID  PositionVCF  \\\n",
       "0  ClinGen:CA10602353,OMIM:191305.0003        12321     11573132   \n",
       "\n",
       "  ReferenceAlleleVCF AlternateAlleleVCF  GenesPerAlleleID  \\\n",
       "0                  C                  T               1.0   \n",
       "\n",
       "                     Category    MC  \n",
       "0  asserted, but not computed  None  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# var id 12321 , 14669\n",
    "c1.execute(\"SELECT * from allele where VariationID = 12321\").fetchdf()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "989d3c6e",
   "metadata": {},
   "source": [
    "# ORG"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "d8b9ad20",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>OrganizationName</th>\n",
       "      <th>OrganizationID</th>\n",
       "      <th>InstitutionType</th>\n",
       "      <th>NumberOfClinVarSubmissions</th>\n",
       "      <th>MaximumReviewStatus</th>\n",
       "      <th>CollectionMethods</th>\n",
       "      <th>ClinicalSignificanceCategoriesSubmitted</th>\n",
       "      <th>NumberOfSubmissionsFromClinicalTesting</th>\n",
       "      <th>NumberOfSubmissionsFromResearch</th>\n",
       "      <th>NumberOfSubmissionsFromLiteratureOnly</th>\n",
       "      <th>NumberOfSubmissionsFromCuration</th>\n",
       "      <th>NumberOfSubmissionsFromPhenotyping</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OMIM; Johns Hopkins University</td>\n",
       "      <td>3</td>\n",
       "      <td>resource</td>\n",
       "      <td>36730</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>literature only</td>\n",
       "      <td>Pathogenic, Likely pathogenic, Uncertain significance, Benign, drug response, other, Affects, association, protective, risk factor, not provided</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>36638</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Baylor Genetics</td>\n",
       "      <td>1006</td>\n",
       "      <td>lab</td>\n",
       "      <td>51195</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>clinical testing, research</td>\n",
       "      <td>Pathogenic, Likely pathogenic, Uncertain significance, Likely benign, Benign</td>\n",
       "      <td>51167</td>\n",
       "      <td>29</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Athena Diagnostics</td>\n",
       "      <td>1012</td>\n",
       "      <td>lab</td>\n",
       "      <td>21962</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>clinical testing</td>\n",
       "      <td>Pathogenic, Likely pathogenic, Uncertain significance, Likely benign, Benign, Likely risk allele</td>\n",
       "      <td>21962</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Sanford Medical Genetics Laboratory; Sanfordhealth</td>\n",
       "      <td>1018</td>\n",
       "      <td>lab</td>\n",
       "      <td>2</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>clinical testing</td>\n",
       "      <td>Pathogenic, Uncertain significance</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Greenwood Genetic Center Diagnostic Laboratories; Greenwood Genetic Center</td>\n",
       "      <td>1019</td>\n",
       "      <td>lab</td>\n",
       "      <td>3920</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>clinical testing, research</td>\n",
       "      <td>Pathogenic, Likely pathogenic, Uncertain significance, Likely benign, Benign</td>\n",
       "      <td>3910</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                             OrganizationName  \\\n",
       "0                                              OMIM; Johns Hopkins University   \n",
       "1                                                             Baylor Genetics   \n",
       "2                                                          Athena Diagnostics   \n",
       "3                          Sanford Medical Genetics Laboratory; Sanfordhealth   \n",
       "4  Greenwood Genetic Center Diagnostic Laboratories; Greenwood Genetic Center   \n",
       "\n",
       "   OrganizationID InstitutionType  NumberOfClinVarSubmissions  \\\n",
       "0               3        resource                       36730   \n",
       "1            1006             lab                       51195   \n",
       "2            1012             lab                       21962   \n",
       "3            1018             lab                           2   \n",
       "4            1019             lab                        3920   \n",
       "\n",
       "                   MaximumReviewStatus           CollectionMethods  \\\n",
       "0       no assertion criteria provided             literature only   \n",
       "1  criteria provided, single submitter  clinical testing, research   \n",
       "2  criteria provided, single submitter            clinical testing   \n",
       "3       no assertion criteria provided            clinical testing   \n",
       "4  criteria provided, single submitter  clinical testing, research   \n",
       "\n",
       "                                                                                                            ClinicalSignificanceCategoriesSubmitted  \\\n",
       "0  Pathogenic, Likely pathogenic, Uncertain significance, Benign, drug response, other, Affects, association, protective, risk factor, not provided   \n",
       "1                                                                      Pathogenic, Likely pathogenic, Uncertain significance, Likely benign, Benign   \n",
       "2                                                  Pathogenic, Likely pathogenic, Uncertain significance, Likely benign, Benign, Likely risk allele   \n",
       "3                                                                                                                Pathogenic, Uncertain significance   \n",
       "4                                                                      Pathogenic, Likely pathogenic, Uncertain significance, Likely benign, Benign   \n",
       "\n",
       "   NumberOfSubmissionsFromClinicalTesting  NumberOfSubmissionsFromResearch  \\\n",
       "0                                       0                                0   \n",
       "1                                   51167                               29   \n",
       "2                                   21962                                0   \n",
       "3                                       2                                0   \n",
       "4                                    3910                               10   \n",
       "\n",
       "   NumberOfSubmissionsFromLiteratureOnly  NumberOfSubmissionsFromCuration  \\\n",
       "0                                  36638                                0   \n",
       "1                                      0                                0   \n",
       "2                                      0                                0   \n",
       "3                                      0                                0   \n",
       "4                                      0                                0   \n",
       "\n",
       "   NumberOfSubmissionsFromPhenotyping  \n",
       "0                                   0  \n",
       "1                                   0  \n",
       "2                                   0  \n",
       "3                                   0  \n",
       "4                                   0  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "c1.execute(\"SELECT * from organization_summary limit 5\").fetchdf()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3bc2f0e4",
   "metadata": {},
   "source": [
    "# citation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "533a49b6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AlleleID</th>\n",
       "      <th>VariationID</th>\n",
       "      <th>CitationSource</th>\n",
       "      <th>CitationID</th>\n",
       "      <th>OrganizationID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>27360</td>\n",
       "      <td>12321</td>\n",
       "      <td>PubMed</td>\n",
       "      <td>19667185</td>\n",
       "      <td>1,3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>27360</td>\n",
       "      <td>12321</td>\n",
       "      <td>PubMed</td>\n",
       "      <td>33116287</td>\n",
       "      <td>508299</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   AlleleID  VariationID CitationSource CitationID OrganizationID\n",
       "0     27360        12321         PubMed   19667185            1,3\n",
       "1     27360        12321         PubMed   33116287         508299"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "\n",
    "c1.execute(\"SELECT * from var_citations where VariationID= 12321 \").fetchdf()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8c07826d",
   "metadata": {},
   "source": [
    "# submission summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "37875678",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>VariationID</th>\n",
       "      <th>ClinicalSignificance</th>\n",
       "      <th>Description</th>\n",
       "      <th>SubmittedPhenotypeInfo</th>\n",
       "      <th>ReportedPhenotypeInfo</th>\n",
       "      <th>ReviewStatus</th>\n",
       "      <th>CollectionMethod</th>\n",
       "      <th>OriginCounts</th>\n",
       "      <th>Submitter</th>\n",
       "      <th>SubmittedGeneSymbol</th>\n",
       "      <th>ExplanationOfInterpretation</th>\n",
       "      <th>ContributesToAggregateClassification</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>12321</td>\n",
       "      <td>Benign</td>\n",
       "      <td>NC_000008.11:g.11573132C&gt;T has an allele frequency of 0.163 in African subpopulation in the gnomAD database, including 128 homozygous occurrences. This variant was annotated at the position 11,468,050 on chromsome 8. In the functional study by MIN6 beta-cells, this variant decreased luciferase expression as compared with control constructs without any insert (PMID: 19667185). This evidence suggests the variant to be classified as benign. ACMG/AMP criteria applied: BA1, BS2, PS3.</td>\n",
       "      <td>Maturity-onset diabetes of the young, type 11</td>\n",
       "      <td>C3150618:Maturity-onset diabetes of the young type 11</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>curation</td>\n",
       "      <td>germline:na</td>\n",
       "      <td>Reproductive Health Research and Development, BGI Genomics</td>\n",
       "      <td>BLK</td>\n",
       "      <td>-</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>12321</td>\n",
       "      <td>Benign</td>\n",
       "      <td>This variant is classified as benign based on ACMG/AMP sequence variant interpretation guidelines (Richards et al. 2015 PMID: 25741868, with internal and published modifications).</td>\n",
       "      <td>BLK-related condition</td>\n",
       "      <td>na:BLK-related disorder</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>clinical testing</td>\n",
       "      <td>germline:na</td>\n",
       "      <td>PreventionGenetics, part of Exact Sciences</td>\n",
       "      <td>BLK</td>\n",
       "      <td>-</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>12321</td>\n",
       "      <td>Benign</td>\n",
       "      <td>While the frequency of the alternate allele in gnoMAD v2.0.2 is 0.146, its frequency in African populations is &gt;5%. This suggests that previous classifications of this variant as pathogenic are in error.</td>\n",
       "      <td>not specified</td>\n",
       "      <td>CN169374:not specified</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>research</td>\n",
       "      <td>germline:na</td>\n",
       "      <td>H3Africa Consortium</td>\n",
       "      <td>BLK</td>\n",
       "      <td>-</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>12321</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>MATURITY-ONSET DIABETES OF THE YOUNG, TYPE 11</td>\n",
       "      <td>C3150618:Maturity-onset diabetes of the young type 11</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>literature only</td>\n",
       "      <td>germline:na</td>\n",
       "      <td>OMIM</td>\n",
       "      <td>BLK</td>\n",
       "      <td>-</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   VariationID ClinicalSignificance  \\\n",
       "0        12321               Benign   \n",
       "1        12321               Benign   \n",
       "2        12321               Benign   \n",
       "3        12321           Pathogenic   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           Description  \\\n",
       "0  NC_000008.11:g.11573132C>T has an allele frequency of 0.163 in African subpopulation in the gnomAD database, including 128 homozygous occurrences. This variant was annotated at the position 11,468,050 on chromsome 8. In the functional study by MIN6 beta-cells, this variant decreased luciferase expression as compared with control constructs without any insert (PMID: 19667185). This evidence suggests the variant to be classified as benign. ACMG/AMP criteria applied: BA1, BS2, PS3.   \n",
       "1                                                                                                                                                                                                                                                                                                                  This variant is classified as benign based on ACMG/AMP sequence variant interpretation guidelines (Richards et al. 2015 PMID: 25741868, with internal and published modifications).   \n",
       "2                                                                                                                                                                                                                                                                                          While the frequency of the alternate allele in gnoMAD v2.0.2 is 0.146, its frequency in African populations is >5%. This suggests that previous classifications of this variant as pathogenic are in error.   \n",
       "3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    -   \n",
       "\n",
       "                          SubmittedPhenotypeInfo  \\\n",
       "0  Maturity-onset diabetes of the young, type 11   \n",
       "1                          BLK-related condition   \n",
       "2                                  not specified   \n",
       "3  MATURITY-ONSET DIABETES OF THE YOUNG, TYPE 11   \n",
       "\n",
       "                                   ReportedPhenotypeInfo  \\\n",
       "0  C3150618:Maturity-onset diabetes of the young type 11   \n",
       "1                                na:BLK-related disorder   \n",
       "2                                 CN169374:not specified   \n",
       "3  C3150618:Maturity-onset diabetes of the young type 11   \n",
       "\n",
       "                          ReviewStatus  CollectionMethod OriginCounts  \\\n",
       "0       no assertion criteria provided          curation  germline:na   \n",
       "1       no assertion criteria provided  clinical testing  germline:na   \n",
       "2  criteria provided, single submitter          research  germline:na   \n",
       "3       no assertion criteria provided   literature only  germline:na   \n",
       "\n",
       "                                                    Submitter  \\\n",
       "0  Reproductive Health Research and Development, BGI Genomics   \n",
       "1                  PreventionGenetics, part of Exact Sciences   \n",
       "2                                         H3Africa Consortium   \n",
       "3                                                        OMIM   \n",
       "\n",
       "  SubmittedGeneSymbol ExplanationOfInterpretation  \\\n",
       "0                 BLK                           -   \n",
       "1                 BLK                           -   \n",
       "2                 BLK                           -   \n",
       "3                 BLK                           -   \n",
       "\n",
       "  ContributesToAggregateClassification  \n",
       "0                                   no  \n",
       "1                                   no  \n",
       "2                                  yes  \n",
       "3                                   no  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "c1.execute(\"SELECT * from submission_summary where VariationID= 12321 \").fetchdf()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "c7a3d20e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>VariationID</th>\n",
       "      <th>AlleleID</th>\n",
       "      <th>NucleotideExpression</th>\n",
       "      <th>NucleotideChange</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>12321</td>\n",
       "      <td>27360</td>\n",
       "      <td>NC_000008.11:g.11573132C&gt;T</td>\n",
       "      <td>g.11573132C&gt;T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   VariationID  AlleleID        NucleotideExpression NucleotideChange\n",
       "0        12321     27360  NC_000008.11:g.11573132C>T    g.11573132C>T"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "c1.execute(\"SELECT * from hgvs4variation where VariationID= 12321 \").fetchdf()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "4bf18437",
   "metadata": {},
   "outputs": [],
   "source": [
    "c1.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "51a2050d",
   "metadata": {},
   "source": [
    "# model train data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "72f4ea28",
   "metadata": {},
   "outputs": [],
   "source": [
    "c2 = duckdb.connect(mtrain)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "ca53fae6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AlleleID</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>ClinicalSignificance</th>\n",
       "      <th>has_MC_nonsense</th>\n",
       "      <th>has_MC_non-coding_transcript_variant</th>\n",
       "      <th>has_MC_missense_variant</th>\n",
       "      <th>has_MC_intron_variant</th>\n",
       "      <th>has_MC_5_prime_UTR_variant</th>\n",
       "      <th>has_MC_splice_donor_variant</th>\n",
       "      <th>has_MC_synonymous_variant</th>\n",
       "      <th>has_MC_splice_acceptor_variant</th>\n",
       "      <th>has_MC_initiator_codon_variant</th>\n",
       "      <th>has_MC_3_prime_UTR_variant</th>\n",
       "      <th>has_MC_no_sequence_alteration</th>\n",
       "      <th>has_MC_stop_lost</th>\n",
       "      <th>has_MC_genic_upstream_transcript_variant</th>\n",
       "      <th>has_MC_genic_downstream_transcript_variant</th>\n",
       "      <th>has_Origin_germline</th>\n",
       "      <th>has_Origin_biparental</th>\n",
       "      <th>has_Origin_unknown</th>\n",
       "      <th>has_Origin_maternal</th>\n",
       "      <th>has_Origin_paternal</th>\n",
       "      <th>has_Origin_inherited</th>\n",
       "      <th>has_Origin_de novo</th>\n",
       "      <th>has_Origin_not applicable</th>\n",
       "      <th>has_Origin_tested-inconclusive</th>\n",
       "      <th>has_Origin_uniparental</th>\n",
       "      <th>has_Origin_not-reported</th>\n",
       "      <th>has_VariantGeneRelation_within single gene</th>\n",
       "      <th>has_VariantGeneRelation_within multiple genes by overlap</th>\n",
       "      <th>has_VariantGeneRelation_asserted, but not computed</th>\n",
       "      <th>has_VariantGeneRelation_near gene, upstream</th>\n",
       "      <th>has_VariantGeneRelation_near gene, downstream</th>\n",
       "      <th>has_VariantGeneRelation_not identified</th>\n",
       "      <th>ref_is_A</th>\n",
       "      <th>ref_is_T</th>\n",
       "      <th>ref_is_G</th>\n",
       "      <th>ref_is_C</th>\n",
       "      <th>alt_is_A</th>\n",
       "      <th>alt_is_T</th>\n",
       "      <th>alt_is_G</th>\n",
       "      <th>alt_is_C</th>\n",
       "      <th>chr_11</th>\n",
       "      <th>chr_6</th>\n",
       "      <th>chr_2</th>\n",
       "      <th>chr_20</th>\n",
       "      <th>chr_10</th>\n",
       "      <th>chr_16</th>\n",
       "      <th>chr_22</th>\n",
       "      <th>chr_15</th>\n",
       "      <th>chr_1</th>\n",
       "      <th>chr_7</th>\n",
       "      <th>chr_8</th>\n",
       "      <th>chr_14</th>\n",
       "      <th>chr_21</th>\n",
       "      <th>chr_5</th>\n",
       "      <th>chr_4</th>\n",
       "      <th>chr_19</th>\n",
       "      <th>chr_3</th>\n",
       "      <th>chr_17</th>\n",
       "      <th>chr_12</th>\n",
       "      <th>chr_18</th>\n",
       "      <th>chr_9</th>\n",
       "      <th>chr_13</th>\n",
       "      <th>chr_MT</th>\n",
       "      <th>chr_Y</th>\n",
       "      <th>chr_X</th>\n",
       "      <th>is_genomic</th>\n",
       "      <th>is_mitochondrial</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>15044</td>\n",
       "      <td>55572</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>15045</td>\n",
       "      <td>55572</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>15049</td>\n",
       "      <td>3077</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>15053</td>\n",
       "      <td>3077</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>15058</td>\n",
       "      <td>3077</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   AlleleID  GeneID  ClinicalSignificance  has_MC_nonsense  \\\n",
       "0     15044   55572                     1                1   \n",
       "1     15045   55572                     1                0   \n",
       "2     15049    3077                     1                0   \n",
       "3     15053    3077                     0                0   \n",
       "4     15058    3077                     1                0   \n",
       "\n",
       "   has_MC_non-coding_transcript_variant  has_MC_missense_variant  \\\n",
       "0                                     1                        0   \n",
       "1                                     1                        1   \n",
       "2                                     1                        1   \n",
       "3                                     0                        0   \n",
       "4                                     0                        1   \n",
       "\n",
       "   has_MC_intron_variant  has_MC_5_prime_UTR_variant  \\\n",
       "0                      0                           0   \n",
       "1                      0                           0   \n",
       "2                      1                           0   \n",
       "3                      1                           0   \n",
       "4                      1                           0   \n",
       "\n",
       "   has_MC_splice_donor_variant  has_MC_synonymous_variant  \\\n",
       "0                            0                          0   \n",
       "1                            0                          0   \n",
       "2                            0                          0   \n",
       "3                            0                          0   \n",
       "4                            0                          0   \n",
       "\n",
       "   has_MC_splice_acceptor_variant  has_MC_initiator_codon_variant  \\\n",
       "0                               0                               0   \n",
       "1                               0                               0   \n",
       "2                               0                               0   \n",
       "3                               0                               0   \n",
       "4                               0                               0   \n",
       "\n",
       "   has_MC_3_prime_UTR_variant  has_MC_no_sequence_alteration  \\\n",
       "0                           0                              0   \n",
       "1                           0                              0   \n",
       "2                           0                              0   \n",
       "3                           0                              0   \n",
       "4                           0                              0   \n",
       "\n",
       "   has_MC_stop_lost  has_MC_genic_upstream_transcript_variant  \\\n",
       "0                 0                                         0   \n",
       "1                 0                                         0   \n",
       "2                 0                                         0   \n",
       "3                 0                                         0   \n",
       "4                 0                                         0   \n",
       "\n",
       "   has_MC_genic_downstream_transcript_variant  has_Origin_germline  \\\n",
       "0                                           0                    1   \n",
       "1                                           0                    1   \n",
       "2                                           0                    1   \n",
       "3                                           0                    1   \n",
       "4                                           0                    1   \n",
       "\n",
       "   has_Origin_biparental  has_Origin_unknown  has_Origin_maternal  \\\n",
       "0                      0                   0                    0   \n",
       "1                      0                   0                    0   \n",
       "2                      1                   1                    0   \n",
       "3                      0                   0                    0   \n",
       "4                      0                   0                    0   \n",
       "\n",
       "   has_Origin_paternal  has_Origin_inherited  has_Origin_de novo  \\\n",
       "0                    0                     0                   0   \n",
       "1                    0                     0                   0   \n",
       "2                    0                     0                   0   \n",
       "3                    0                     0                   0   \n",
       "4                    0                     0                   0   \n",
       "\n",
       "   has_Origin_not applicable  has_Origin_tested-inconclusive  \\\n",
       "0                          0                               0   \n",
       "1                          0                               0   \n",
       "2                          0                               0   \n",
       "3                          0                               0   \n",
       "4                          0                               0   \n",
       "\n",
       "   has_Origin_uniparental  has_Origin_not-reported  \\\n",
       "0                       0                        0   \n",
       "1                       0                        0   \n",
       "2                       0                        0   \n",
       "3                       0                        0   \n",
       "4                       0                        0   \n",
       "\n",
       "   has_VariantGeneRelation_within single gene  \\\n",
       "0                                           1   \n",
       "1                                           1   \n",
       "2                                           0   \n",
       "3                                           1   \n",
       "4                                           1   \n",
       "\n",
       "   has_VariantGeneRelation_within multiple genes by overlap  \\\n",
       "0                                                         0   \n",
       "1                                                         0   \n",
       "2                                                         1   \n",
       "3                                                         0   \n",
       "4                                                         0   \n",
       "\n",
       "   has_VariantGeneRelation_asserted, but not computed  \\\n",
       "0                                                   0   \n",
       "1                                                   0   \n",
       "2                                                   0   \n",
       "3                                                   0   \n",
       "4                                                   0   \n",
       "\n",
       "   has_VariantGeneRelation_near gene, upstream  \\\n",
       "0                                            0   \n",
       "1                                            0   \n",
       "2                                            0   \n",
       "3                                            0   \n",
       "4                                            0   \n",
       "\n",
       "   has_VariantGeneRelation_near gene, downstream  \\\n",
       "0                                              0   \n",
       "1                                              0   \n",
       "2                                              0   \n",
       "3                                              0   \n",
       "4                                              0   \n",
       "\n",
       "   has_VariantGeneRelation_not identified  ref_is_A  ref_is_T  ref_is_G  \\\n",
       "0                                       0         0         0         0   \n",
       "1                                       0         1         0         0   \n",
       "2                                       0         0         0         0   \n",
       "3                                       0         0         0         1   \n",
       "4                                       0         1         0         0   \n",
       "\n",
       "   ref_is_C  alt_is_A  alt_is_T  alt_is_G  alt_is_C  chr_11  chr_6  chr_2  \\\n",
       "0         1         0         1         0         0       1      0      0   \n",
       "1         0         0         0         1         0       1      0      0   \n",
       "2         1         0         0         1         0       0      1      0   \n",
       "3         0         1         0         0         0       0      1      0   \n",
       "4         0         0         0         0         1       0      1      0   \n",
       "\n",
       "   chr_20  chr_10  chr_16  chr_22  chr_15  chr_1  chr_7  chr_8  chr_14  \\\n",
       "0       0       0       0       0       0      0      0      0       0   \n",
       "1       0       0       0       0       0      0      0      0       0   \n",
       "2       0       0       0       0       0      0      0      0       0   \n",
       "3       0       0       0       0       0      0      0      0       0   \n",
       "4       0       0       0       0       0      0      0      0       0   \n",
       "\n",
       "   chr_21  chr_5  chr_4  chr_19  chr_3  chr_17  chr_12  chr_18  chr_9  chr_13  \\\n",
       "0       0      0      0       0      0       0       0       0      0       0   \n",
       "1       0      0      0       0      0       0       0       0      0       0   \n",
       "2       0      0      0       0      0       0       0       0      0       0   \n",
       "3       0      0      0       0      0       0       0       0      0       0   \n",
       "4       0      0      0       0      0       0       0       0      0       0   \n",
       "\n",
       "   chr_MT  chr_Y  chr_X  is_genomic  is_mitochondrial  \n",
       "0       0      0      0           1                 0  \n",
       "1       0      0      0           1                 0  \n",
       "2       0      0      0           1                 0  \n",
       "3       0      0      0           1                 0  \n",
       "4       0      0      0           1                 0  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "c2.execute(\"select * from allele limit 5\").fetchdf()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "5bd56ecd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('model_train_data',\n",
       "  'main',\n",
       "  'allele',\n",
       "  ['AlleleID',\n",
       "   'GeneID',\n",
       "   'ClinicalSignificance',\n",
       "   'has_MC_nonsense',\n",
       "   'has_MC_non-coding_transcript_variant',\n",
       "   'has_MC_missense_variant',\n",
       "   'has_MC_intron_variant',\n",
       "   'has_MC_5_prime_UTR_variant',\n",
       "   'has_MC_splice_donor_variant',\n",
       "   'has_MC_synonymous_variant',\n",
       "   'has_MC_splice_acceptor_variant',\n",
       "   'has_MC_initiator_codon_variant',\n",
       "   'has_MC_3_prime_UTR_variant',\n",
       "   'has_MC_no_sequence_alteration',\n",
       "   'has_MC_stop_lost',\n",
       "   'has_MC_genic_upstream_transcript_variant',\n",
       "   'has_MC_genic_downstream_transcript_variant',\n",
       "   'has_Origin_germline',\n",
       "   'has_Origin_biparental',\n",
       "   'has_Origin_unknown',\n",
       "   'has_Origin_maternal',\n",
       "   'has_Origin_paternal',\n",
       "   'has_Origin_inherited',\n",
       "   'has_Origin_de novo',\n",
       "   'has_Origin_not applicable',\n",
       "   'has_Origin_tested-inconclusive',\n",
       "   'has_Origin_uniparental',\n",
       "   'has_Origin_not-reported',\n",
       "   'has_VariantGeneRelation_within single gene',\n",
       "   'has_VariantGeneRelation_within multiple genes by overlap',\n",
       "   'has_VariantGeneRelation_asserted, but not computed',\n",
       "   'has_VariantGeneRelation_near gene, upstream',\n",
       "   'has_VariantGeneRelation_near gene, downstream',\n",
       "   'has_VariantGeneRelation_not identified',\n",
       "   'ref_is_A',\n",
       "   'ref_is_T',\n",
       "   'ref_is_G',\n",
       "   'ref_is_C',\n",
       "   'alt_is_A',\n",
       "   'alt_is_T',\n",
       "   'alt_is_G',\n",
       "   'alt_is_C',\n",
       "   'chr_11',\n",
       "   'chr_6',\n",
       "   'chr_2',\n",
       "   'chr_20',\n",
       "   'chr_10',\n",
       "   'chr_16',\n",
       "   'chr_22',\n",
       "   'chr_15',\n",
       "   'chr_1',\n",
       "   'chr_7',\n",
       "   'chr_8',\n",
       "   'chr_14',\n",
       "   'chr_21',\n",
       "   'chr_5',\n",
       "   'chr_4',\n",
       "   'chr_19',\n",
       "   'chr_3',\n",
       "   'chr_17',\n",
       "   'chr_12',\n",
       "   'chr_18',\n",
       "   'chr_9',\n",
       "   'chr_13',\n",
       "   'chr_MT',\n",
       "   'chr_Y',\n",
       "   'chr_X',\n",
       "   'is_genomic',\n",
       "   'is_mitochondrial'],\n",
       "  ['BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT',\n",
       "   'BIGINT'],\n",
       "  False)]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "c2.execute(\"describe\").fetchall()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "04df524f",
   "metadata": {},
   "outputs": [],
   "source": [
    "c2.close()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4426990b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# 15044,\"nonsense,non-coding_transcript_variant\",germline,C,T,11,within single gene,g,Pathogenic,0.999173104763031"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv (3.13.3)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
