{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "295a5867",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import duckdb \n",
    "import pickle\n",
    "from tqdm import tqdm\n",
    "import shutil"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e638dc3e",
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.set_option('display.max_colwidth', None)  \n",
    "pd.set_option('display.expand_frame_repr', False)  \n",
    "pd.set_option('display.max_columns', None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c1118fcb",
   "metadata": {},
   "outputs": [],
   "source": [
    "base_path = \"/kaggle/input/preprocessed-capstone-data-3/data\"\n",
    "\n",
    "gene_df = pd.read_csv(f\"{base_path}/gene_df.tsv\", sep=\"\\t\")\n",
    "allele_df = pd.read_csv(f\"{base_path}/allele_df.tsv\", sep=\"\\t\")\n",
    "hgvs4variation_df = pd.read_csv(f\"{base_path}/hgvs4variation_df.tsv\", sep=\"\\t\")\n",
    "cross_references_df = pd.read_csv(f\"{base_path}/cross_references_df.tsv\", sep=\"\\t\")\n",
    "organization_summary_df = pd.read_csv(f\"{base_path}/organization_summary_df_1.tsv\", sep=\"\\t\")\n",
    "submission_summary_df = pd.read_csv(f\"{base_path}/submission_summary_df.tsv\", sep=\"\\t\")\n",
    "summary_of_conflicting_interpretations_df = pd.read_csv(f\"{base_path}/summary_of_conflicting_interpretations_df.tsv\", sep=\"\\t\")\n",
    "var_citations_df = pd.read_csv(f\"{base_path}/var_citations_df.tsv\", sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "680e60f6",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_list = []\n",
    "for _, i in gene_df.iterrows():\n",
    "    gene_disease = i[\"GeneLevelDisease\"]\n",
    "    \n",
    "    if gene_disease.strip().lower() in [\"\", \"no gene-disease association has been submitted to clinvar for this gene\"]:\n",
    "        disease_sentence = \"No gene-disease association has been submitted to ClinVar for this gene\"\n",
    "    else:\n",
    "        disease_sentence = f\"This gene has been associated with: {gene_disease}\"\n",
    "\n",
    "    content_lines = [\n",
    "        f\"{i['GeneSymbol']} is the gene symbol, and its full name is {i['GeneName']}.\",\n",
    "        f\"{disease_sentence}.\".strip()\n",
    "    ]\n",
    "    doc = {\n",
    "        'type': 'gene',\n",
    "        'metadata': {\n",
    "            'GeneID': i[\"GeneID\"]\n",
    "        },\n",
    "        'page_content': \"\\n\".join(content_lines)\n",
    "    }\n",
    "\n",
    "    gene_list.append(doc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "200c0558",
   "metadata": {},
   "outputs": [],
   "source": [
    "allele_list = []\n",
    "for _, i in tqdm(allele_df.iterrows(), total=len(allele_df), desc=\"Processing Alleles\"):\n",
    "    doc = {\n",
    "        'type': 'allele',\n",
    "        'metadata': {\n",
    "            'AlleleID': i[\"AlleleID\"],\n",
    "            'GeneID': i[\"GeneID\"],\n",
    "            'VariationID': i[\"VariationID\"]\n",
    "        },\n",
    "        'page_content': f\"\"\"The allele is linked to the following phenotypes: {i['PhenotypeList']} with clinical significance as {i['ClinicalSignificance']}.\n",
    "Testing for this variant is{' ' if i['TestedInGTR'] == 'Y' else ' not '}registered in GTR.\n",
    "\"\"\".strip()\n",
    "    }\n",
    "    allele_list.append(doc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "258aeb35",
   "metadata": {},
   "outputs": [],
   "source": [
    "organization_summary_list = []\n",
    "for _, i in tqdm(organization_summary_df.iterrows(), total=len(organization_summary_df)):\n",
    "    doc = {\n",
    "        'type': 'organization_summary',\n",
    "        'metadata': {\n",
    "            'OrganizationID': i[\"OrganizationID\"]\n",
    "        },\n",
    "        'page_content': f\"\"\"{i['OrganizationName']} is an organization of type \"{i['InstitutionType']}\".\n",
    "The highest review status achieved is: {i['MaximumReviewStatus']}.\n",
    "Submission collection methods include: {i['CollectionMethods']}.\n",
    "The clinical significance categories submitted by this organization include: {i['ClinicalSignificanceCategoriesSubmitted']}.\n",
    "        \"\"\".strip()\n",
    "    }\n",
    "\n",
    "    organization_summary_list.append(doc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d86edc7f",
   "metadata": {},
   "outputs": [],
   "source": [
    "submission_summary_list = []\n",
    "for _, i in tqdm(submission_summary_df.iterrows(), total=len(submission_summary_df)):\n",
    "    doc = {\n",
    "        'type': 'submission_summary',\n",
    "        'metadata': {\n",
    "            'VariationID': i[\"VariationID\"]\n",
    "        },\n",
    "        'page_content': f\"\"\"\n",
    "This submission was made by: {i['Submitter']}.\n",
    "\n",
    "The clinical significance of the variant in this submission is classified as: {i['ClinicalSignificance']}.\n",
    "Review status for this interpretation is: {i['ReviewStatus']}.\n",
    "\n",
    "The condition was interpreted based on the following submitted phenotype information: {i['SubmittedPhenotypeInfo']}.\n",
    "ClinVar reports this using the MedGen terms: {i['ReportedPhenotypeInfo']}.\n",
    "\n",
    "Collection method used for this data: {i['CollectionMethod']}.\n",
    "Reported origin(s) and number of observations: {i['OriginCounts']}.\n",
    "\n",
    "Explanation provided for the interpretation: {i['ExplanationOfInterpretation']}.\n",
    "\n",
    "Contributes to aggregate classification: {i['ContributesToAggregateClassification']}.\n",
    "\n",
    "Additional description provided by the submitter:\n",
    "{i['Description']}\n",
    "        \"\"\".strip()\n",
    "    }\n",
    "\n",
    "    submission_summary_list.append(doc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "753a959c",
   "metadata": {},
   "outputs": [],
   "source": [
    "summary_of_conflicting_interpretations_list = []\n",
    "for _, i in tqdm(summary_of_conflicting_interpretations_df.iterrows(), total=len(summary_of_conflicting_interpretations_df)):\n",
    "\n",
    "    doc = {\n",
    "        'type': 'summary_of_conflicting_interpretations',\n",
    "        'metadata': {\n",
    "            'VariationID': i[\"VariationID\"]\n",
    "        },\n",
    "        'page_content': f\"\"\"\n",
    "Two submitters have provided differing interpretations for the variant described as: {i['ClinVar_Preferred']}.\n",
    "\n",
    "Submitter 1: {i['Submitter1']}\n",
    "- Clinical significance: {i['Submitter1_ClinSig']}\n",
    "- Review status: {i['Submitter1_ReviewStatus']}\n",
    "- Interpretation description: {i['Submitter1_Description']}\n",
    "- Method(s) used: {i['Submitter1_Method']}\n",
    "\n",
    "Submitter 2: {i['Submitter2']}\n",
    "- Clinical significance: {i['Submitter2_ClinSig']}\n",
    "- Review status: {i['Submitter2_ReviewStatus']}\n",
    "- Interpretation description: {i['Submitter2_Description']}\n",
    "- Method(s) used: {i['Submitter2_Method']}\n",
    "\n",
    "Conflict reported: {i['Conflict_Reported']}\n",
    "Rank difference between interpretations: {i['Rank_diff']}\n",
    "Variant type: {i['Variant_type']}\n",
    "        \"\"\".strip()\n",
    "    }\n",
    "\n",
    "    summary_of_conflicting_interpretations_list.append(doc)"
   ]
  }
 ],
 "metadata": {
  "language_info": {
   "name": "python"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
